torch/_inductor/ir.py

```diff
@@ -6528,6 +6528,71 @@ def _has_aliased_buffers(buffers: Sequence[IRNode]) -> bool:
     return len(OrderedSet(id(buffer) for buffer in buffers)) < len(buffers)
 
 
+@ir_dataclass
+class InvokeSubgraph(ExternKernel):
+    subgraph: Optional[Subgraph] = None
+    operands: Optional[List[TensorBox]] = None
+    outputs: Optional[List[MultiOutput]] = None
+
+    def __init__(
+        self, subgraph: Subgraph, operands: List[TensorBox], layout: MultiOutputLayout
+    ):
+        super().__init__(
+            name=None,
+            layout=layout,
+            inputs=operands,
+        )
+        self.subgraph = subgraph
+        self.name = V.graph.register_buffer(self)
+        V.graph.register_operation(self)
+
+    @classmethod
+    def create(cls, subgraph: Subgraph, operands):
+        operands = [cls.realize_input(x) for x in operands]
+
+        fx_operands = V.graph.current_node.args[-1]
+        fake_operands = [x.meta["val"] for x in fx_operands]  # type: ignore[union-attr]
```

Comment:
inductor is not required to match strides for all intermediaries. We could try to force strides to be equal for these inputs, or we can update the fake_operands to match the strides of the operands input. Generally, it's better to let inductor have flexibility. I guess this also an existing issue with Cond and While loop cc @aakhundov .

If the strides are potentially changing, we may not always be able to reuse the subgraph lowering and would need to do some checking.

Reply:
This is also a reason to implement assert_sizes_strides haha - would manifest as silent incorrectness rn if we had divergence

Reply:
Yes, Horace also mentioned. I am planning to add asserts back for the subgraph

Reply:
Added code to match the stride. Please check.